crates/aptos-faucet/README.md

```diff
@@ -1,56 +1,107 @@
-The response HTTP body is hex encoded bytes of BCS encoded `Vec<aptos_types::transaction::SignedTransaction>`.
+Then build the tap as normal (from the root of the repo):
+```
+cargo build
+```
 
-Decode Example ([source code generator](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/aptos-sdk-builder)):
+## Testing
+If you want to run the tests manually, follow these steps. Note that this is **not necessary** for release safety as the tests are run as part of continuous integration (CI) already.
 
-``` python
-  de = bcs.BcsDeserializer(bytes.fromhex(response.text))
-  length = de.deserialize_len()
+For the test suite to pass, you must spin up a local testnet, notably without a faucet running (since we're testing the faucet here):
+```
+cargo run -p aptos -- node run-local-testnet --force-restart --assume-yes
+```
 
-  txns = []
-  for i in range(length):
-    txns.push(de.deserialize_any(aptos_types.SignedTransaction))
+You must then copy the mint key for that local testnet to the location the tests expect:
+```
+cp ~/.aptos/testnet/mint.key /tmp
+```
 
+As well as spin up a local Redis 6 ([installation guide](https://redis.io/docs/getting-started/)):
+```
+redis-server
+redis-cli flushall
 ```
 
-You should retry the mint API call if the transaction execution fails.
+Finally you can run the tests:
+```
+cargo test -p aptos-faucet-core --features integration-tests
+```
 
+## Validating configs
+To ensure all the configs are valid, run this:
+```
+cd crates/aptos-faucet/configs
+ls . | xargs -I@ cargo run -p aptos-faucet-service -- validate-config -c @
+```
 
-## Example
+## Generating the specs
+From the root:
+```
+cargo run -- generate-openapi -o doc/spec.yaml -f yaml
+cargo run -- generate-openapi -o doc/spec.json -f json
+```
 
-```bash
-curl -X POST http://faucet.testnet.aptoslabs.com/mint\?amount\=1000000\&pub_key\=459c77a38803bd53f3adee52703810e3a74fd7c46952c497e75afb0a7932586d\&return_txns\=true
-01000000000000000000000000000000dd05a600000000000001e001a11ceb0b010000000701000202020403061004160205181d0735600895011000000001010000020001000003020301010004010300010501060c0108000506080005030a020a020005060c05030a020a020109000b4469656d4163636f756e741257697468647261774361706162696c6974791b657874726163745f77697468647261775f6361706162696c697479087061795f66726f6d1b726573746f72655f77697468647261775f6361706162696c69747900000000000000000000000000000001010104010c0b0011000c050e050a010a020b030b0438000b051102020107000000000000000000000000000000010358555303585553000403a74fd7c46952c497e75afb0a7932586d0140420f00000000000400040040420f00000000000000000000000000035855532a610f6000000000020020056244e7bf776e471d818dc18fdf7b8833c5439ac9a96e126f8f32c7bc7c14b64026a2c45c8e4066c661dc4f36baa6ad61499999b548b9f63ad15853660c408cedec3078b7773a829ec48de8b04291cd11530734b2f91d5e42f35a4c6378cb7c09
+## Generating the TS client
+From `ts-client`:
+```
+yarn generate-client
 ```
+
+## Internal
```

Comment:
I highly recommend we remove the Internal section at the bottom!

Reply:
Well it's handy for us to have these backlinks but sure 😛